index.d.ts

```diff
@@ -0,0 +1,240 @@
+// Type definitions for prosemirror-tables 0.8
+// Project: https://github.com/ProseMirror/prosemirror-tables
+// Definitions by: Oscar Wallhult <https://github.com/superchu>
+//                 Eduard Shvedai <https://github.com/eshvedai>
+//                 Patrick Simmelbauer <https://github.com/patsimm>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.3
+import { EditorState, Plugin, SelectionRange, Transaction, PluginKey } from 'prosemirror-state';
+import { Node as ProsemirrorNode, NodeSpec, Slice, ResolvedPos, Schema, NodeType } from 'prosemirror-model';
+import { NodeView } from 'prosemirror-view';
+
+export interface TableEditingOptions {
+  allowTableNodeSelection?: boolean;
+}
+
+export interface TableNodesOptions {
+  tableGroup?: string;
+  cellContent: string;
+  cellAttributes: { [key: string]: CellAttributes };
+}
+
+export type getFromDOM = (dom: Element) => any;
+export type setDOMAttr = (value: any, attrs: any) => any;
+
+export interface CellAttributes {
+  default: any;
+  getFromDOM?: getFromDOM;
+  setDOMAttr?: setDOMAttr;
+}
+
+export interface TableNodes {
+  table: NodeSpec;
+  table_row: NodeSpec;
+  table_cell: NodeSpec;
+  table_header: NodeSpec;
+}
+
+export function tableNodes(options: TableNodesOptions): TableNodes;
+
+export interface CellSelectionJSON {
+  type: string;
+  anchor: number;
+  head: number;
+}
+
+export class CellSelection<S extends Schema = any> {
+  constructor($anchorCell: ResolvedPos<S>, $headCell?: ResolvedPos<S>);
+
+  from: number;
+  to: number;
+  $from: ResolvedPos<S>;
+  $to: ResolvedPos<S>;
+  anchor: number;
+  head: number;
+  $anchor: ResolvedPos<S>;
+  $head: ResolvedPos<S>;
+  $anchorCell: ResolvedPos<S>;
+  $headCell: ResolvedPos<S>;
+  empty: boolean;
+  ranges: Array<SelectionRange<S>>;
+
+  map(doc: ProsemirrorNode<S>, mapping: any): any;
+  content(): Slice<S>;
+  replace(tr: Transaction<S>, content: Slice<S>): void;
+  replaceWith(tr: Transaction<S>, node: ProsemirrorNode<S>): void;
+  forEachCell(f: (node: ProsemirrorNode<S>, pos: number) => void): void;
+  isRowSelection(): boolean;
+  isColSelection(): boolean;
+  eq(other: any): boolean;
```

Comment:
can we use a more definitive type for `other`?

Reply:
Sure, let me fix it. I just copy the file from DefinitelyTyped and I didn't check it 🙈